From akpm@linux-foundation.org  Thu Oct 21 11:39:13 2010
Message-Id: <201010202301.o9KN1pnO009775@imap1.linux-foundation.org>
Subject: USB: gadget: amd5536udc.c: fix error path
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org,
        rahul.ruikar@gmail.com, dahlmann.thomas@arcor.de, david-b@pacbell.net
From: Rahul Ruikar <rahul.ruikar@gmail.com>
Date: Wed, 20 Oct 2010 16:01:51 -0700

From: Rahul Ruikar <rahul.ruikar@gmail.com>

In function udc_probe() call put_device() when device_register() fails.

Signed-off-by: Rahul Ruikar <rahul.ruikar@gmail.com>
Acked-by: Thomas Dahlmann <dahlmann.thomas@arcor.de>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/amd5536udc.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/usb/gadget/amd5536udc.c
+++ b/drivers/usb/gadget/amd5536udc.c
@@ -3383,8 +3383,10 @@ static int udc_probe(struct udc *dev)
 	udc = dev;
 
 	retval = device_register(&dev->gadget.dev);
-	if (retval)
+	if (retval) {
+		put_device(&dev->gadget.dev);
 		goto finished;
+	}
 
 	/* timer init */
 	init_timer(&udc_timer);
